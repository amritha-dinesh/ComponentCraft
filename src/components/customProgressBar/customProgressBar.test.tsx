import React from "react";
import { render } from "@testing-library/react-native";
import CustomProgressBar from "./customProgressBar";

describe("CustomProgressBar", () => {
  it("should match the snapshot", () => {
    const { toJSON } = render(<CustomProgressBar progress={0} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId } = render(<CustomProgressBar progress={0} />);
    const customProgressBarComponent = getByTestId("custom-progress-bar");
    expect(customProgressBarComponent).toBeDefined();
  });

  it("should have the default border radius when not specified", () => {
    const { getByTestId } = render(<CustomProgressBar progress={0} />);
    const customProgressBarComponent = getByTestId("custom-progress-bar");
    const containerElement = customProgressBarComponent.props.style[1];
    expect(containerElement).toHaveProperty("borderRadius", 5);
  });

  it("should apply custom width and height when specified", () => {
    const { getByTestId } = render(
      <CustomProgressBar progress={0} width={200} height={20} />
    );
    const customProgressBarComponent = getByTestId("custom-progress-bar");
    const containerElement = customProgressBarComponent.props.style[1];
    const { width, height } = containerElement;
    expect(width).toBe(200);
    expect(height).toBe(20);
  });
});
