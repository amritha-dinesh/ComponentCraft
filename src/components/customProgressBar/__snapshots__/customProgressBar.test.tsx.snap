// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomProgressBar should match the snapshot 1`] = `
<RCTSafeAreaView>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "backgroundColor": "#dcedc8",
          "borderRadius": 5,
          "height": 10,
          "width": "100%",
        },
      ]
    }
    testID="custom-progress-bar"
  >
    <View
      style={
        [
          {
            "height": "100%",
          },
          {
            "backgroundColor": "#2e7d32",
            "width": "0%",
          },
        ]
      }
    />
  </View>
</RCTSafeAreaView>
`;
